relayer/paths/path_end.go

```diff
@@ -0,0 +1,108 @@
+package paths
+
+import "github.com/cosmos/relayer/v2/relayer/ibc"
+
+type PathEnd struct {
+	ChainID      string
+	ClientID     string
+	ConnectionID string
+
+	// probably just allowlist to start
+	// can only provide one. panic if both are provided
+	AllowList []ibc.ChannelKey // only use these if provided
+	BlockList []ibc.ChannelKey // use everything except these if provided
+}
+
+func NewPathEnd(chainID string, clientID string, connectionID string, allowList []ibc.ChannelKey, blockList []ibc.ChannelKey) PathEnd {
+	if len(allowList) > 0 && len(blockList) > 0 {
+		panic("only one of allowlist or blocklist are allowed")
+	}
+	return PathEnd{
+		ChainID:      chainID,
+		ClientID:     clientID,
+		ConnectionID: connectionID,
+		AllowList:    allowList,
+		BlockList:    blockList,
+	}
+}
+
+// expanded for readability
+func (pe PathEnd) ShouldRelayChannel(channelKey ibc.ChannelKey) bool {
+	if len(pe.AllowList) > 0 {
+		for _, allowedChannel := range pe.AllowList {
```

Comment:
Quite a bit happening here. 

For a future PR, I can see this abstracted to an authorizaton package or similar that's unit tested.

Reply:
agreed, it would be good to clean up the channel filter logic